### ISSUE TITLE
Should stacking be limited to two levels?

### ISSUE THREAD

**Issue by johnhbenetech** [benetech/MathEditor#152]
Currently you can seemingly add infinite levels of stacking. After about 3-4 levels things get very hairy. The input completely escapes the entry box. When you add the go to add the step, the display above gets pretty messed up and it seems to often add 2 steps instead of 1.

[https://user-images.githubusercontent.com/12927553/41332396-8bf950a8-6e92-11e8-80d4-75ec1a85d455.png](https://user-images.githubusercontent.com/12927553/41332396-8bf950a8-6e92-11e8-80d4-75ec1a85d455.png)

Aha! Link: https://diagramlabs.aha.io/features/B-340

**Comment by johnhbenetech** [comment]
@sueannma @NSoiffer Do you guys have opinions on what limitations we should add?

**Comment by sueannma** [comment]
@NSoiffer - I would say let's start with two levels for algebra, but we may need a few more if we are to support some of the more complex examples that were shared at the AIM workshop (things that more closely resembled a multi-leveled matrix-like format).

I would prefer to stick with two tiers now until we encounter specific scenarios where additional rows are required - thoughts?

**Comment by johnhbenetech** [comment]
I'm leaning towards starting with two as well. 

@NSoiffer does the stacking currently utlize a LaTeX matrix? or is it something else entirely?

If it doesnt, we could add a matrix option in the future to handle a more generalized case.

**Comment by NSoiffer** [comment]
Yes, stacking makes use of the TeX command \begin{array}...end{array}.
These are basically the same as \begin{matrix}...\end{matrix} except the
later adds parens around the contents.
